datafusion/src/physical_plan/expressions/binary.rs

```diff
@@ -339,6 +340,63 @@ macro_rules! boolean_op {
     }};
 }
 
+/// Invoke a regexp_match kernel on a pair of arrays
+macro_rules! regexp_match_op {
+    ($LEFT:expr, $RIGHT:expr, $NOT:expr, $CASE:expr) => {{
+        match $LEFT.data_type() {
+            DataType::Utf8 => {
```

Comment:
`LargeUtf8`/`StringArray` would be nice to suport too I think.

Reply:
I have added `LargeUtf8` support, PTAL @Dandandan